Backported Herwig++

mharrend opened the pull request:
Backported Herwig++ due to being asked by @bendavid

Commit:
Merge pull request #4 from cms-sw/CMSSW_5_3_X

Merge newest Cmssw 5 3 x

Commit:
Backported ThePEG interface from CMSSW 7.5 / 8_0_X
Still have to add the additional configs, but first configs are updated

Commit:
HerwigPP: Backported corresponding Generator configs from CMSSW_8_0_X

cmsbuild commented:
A new Pull Request was created by @mharrend (Marco A. Harrendorf) for CMSSW_5_3_X.

Backported Herwig++

It involves the following packages:

Configuration/Generator
GeneratorInterface/ThePEGInterface


@vciulli, @covarell, @thuer, @cmsbuild, @bendavid, @govoni can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald, @mkirsano this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' or '@cmsbuild, please test' in the first line of a comment.
@smuzaffar you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

covarell commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_5_3_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1 
Tested at: 24bbd9e6e5afd09ad94ce2f4ee43d0abae6baf4f
 I found an error when building: 
 
 <pre>>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_5_3_X_2015-11-09-2300/src/GeneratorInterface/ThePEGInterface/src/ThePEGInterface.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_5_3_X_2015-11-09-2300/src/GeneratorInterface/ThePEGInterface/src/RandomEngineGlue.cc 
>> Leaving Package Configuration/Generator
>> Package Configuration/Generator built
>> Subsystem Configuration built
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_5_3_X_2015-11-09-2300/src/GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc:53:22: error: 'virtual void ThePEGHadronizer::doSetRandomEngine(CLHEP::HepRandomEngine*)' marked override, but does not override
>> Building shared library tmp/slc6_amd64_gcc472/src/GeneratorInterface/ThePEGInterface/src/GeneratorInterfaceThePEGInterface/libGeneratorInterfaceThePEGInterface.so
gmake: *** [tmp/slc6_amd64_gcc472/src/GeneratorInterface/ThePEGInterface/plugins/GeneratorInterfaceThePEGHadronizerPlugins/ThePEGHadronizer.o] Error 1
Copying tmp/slc6_amd64_gcc472/src/GeneratorInterface/ThePEGInterface/src/GeneratorInterfaceThePEGInterface/libGeneratorInterfaceThePEGInterface.so to productstore area:
Leaving library rule at GeneratorInterface/ThePEGInterface
>> Building edm plugin tmp/slc6_amd64_gcc472/src/GeneratorInterface/ThePEGInterface/plugins/GeneratorInterfaceThePEGHadronizerPlugins/libGeneratorInterfaceThePEGHadronizerPlugins.so
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12295/9651/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_5_3_X IBs (but tests are reportedly failing). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

covarell commented:
This looks like a problem in compiling @mharrend

mharrend commented on GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc:
```diff
@@ -41,15 +49,22 @@ class ThePEGHadronizer : public ThePEGInterface, public gen::BaseHadronizer {
 
     private:
+
+        virtual void doSetRandomEngine(CLHEP::HepRandomEngine* v) override { setPEGRandomEngine(v); }
```
It seems that the corresponding CLHEP::HepRandomEngine* v function is not available in CMSSW 5_3_X.

mharrend replied:
I will remove this line and hope it fixes our problem with the compilation.

In the worst case this could lead to using the ThePEG random number generator with the same initial seed for each job. But this should be no problem since we will use it only for showering LHE files.

Commit:
Update ThePEGHadronizer.cc

Tried to fix compilation error which is most likely caused by not finding the CLHEP::HepRandomEngine function.

See line notes for further remarks.

cmsbuild commented:
Pull request #12295 was updated. @vciulli, @covarell, @thuer, @cmsbuild, @bendavid, @davidlange6, @govoni can you please check and sign again.

mharrend commented:
please test

bendavid commented:
Hmm, even when showering lhe files it's still not good to reuse the same random seeds.

There was a proper way of handling this in 53x as well.
Suggest to diff eg the Pythia8Interface between 53x and 71x to understand the proper way to handle it in 53x.

covarell commented:
Any news here?

covarell commented:
We are requesting a new 53X soon, so it would be good if this is finalized...

mharrend commented:
Hello Roberto,

I was planning to do this after fixing the problem with the lhapdf and the 
Herwig 7 PR in Cmssw 8X.
But I will also look into this from Thursday on.
Unfortunately, the last weeks I was busy with the ttH preapproval and until 
tomorrow evening I am attending a conference.
Thursday and Friday I planned anyway to look only into Herwig to get Herwig 
7 out with the first Cmssw 8X release.

Best regards,

Marco

Am Dienstag, 23. Februar 2016 09:57:55 CET schrieb Roberto Covarelli 
<notifications@github.com>:
> We are requesting a new 53X soon, so it would be good if this 
> is finalized... 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/12295#issuecomment-187606913
>


-- 
Marco A. Harrendorf
KIT - IEKP

davidlange6 commented:
@mharrend - is this PR still relevant?

perrozzi commented:
@davidlange6 you can close it, thanks
